stream-chat-android-client/src/main/java/io/getstream/chat/android/client/parser/ChatParser.kt

```diff
@@ -1,15 +1,57 @@
 package io.getstream.chat.android.client.parser
 
+import io.getstream.chat.android.client.errors.ChatError
+import io.getstream.chat.android.client.errors.ChatErrorCode
 import io.getstream.chat.android.client.errors.ChatNetworkError
+import io.getstream.chat.android.client.logger.ChatLogger
+import io.getstream.chat.android.client.socket.ErrorResponse
 import io.getstream.chat.android.client.utils.Result
 import okhttp3.Response
 import retrofit2.Retrofit
 
 internal interface ChatParser {
 
+    private val TAG: String
+        get() = ChatParser::class.java.simpleName
+
     fun toJson(any: Any): String
     fun <T : Any> fromJson(raw: String, clazz: Class<T>): T
-    fun <T : Any> fromJsonOrError(raw: String, clazz: Class<T>): Result<T>
-    fun toError(okHttpResponse: Response): ChatNetworkError
     fun configRetrofit(builder: Retrofit.Builder): Retrofit.Builder
+
+    fun <T : Any> fromJsonOrError(raw: String, clazz: Class<T>): Result<T> {
+        return try {
+            Result(fromJson(raw, clazz))
+        } catch (t: Throwable) {
+            Result(ChatError("fromJsonOrError error parsing of $clazz into $raw", t))
+        }
+    }
+
+    fun toError(okHttpResponse: Response): ChatNetworkError {
```

Comment:
Still can't get used to some implementation in interfaces )

Reply:
It's super handy sometimes!